spring-batch-infrastructure/src/main/java/org/springframework/batch/item/avro/builder/AvroItemReaderBuilder.java

```diff
@@ -170,10 +170,6 @@ public AvroItemReader<T> build() {
 
 		avroItemReader.setSaveState(this.saveState);
 
-		if (this.saveState) {
-			Assert.state(StringUtils.hasText(this.name), "A name is required when saveState is set to true.");
-		}
-
```

Comment:
> remove useless assert state

This assertion is not useless. Here is a case where the name could end up being null or empty even with a default name in the constructor:

```java
AvroItemReader avroItemReader = new AvroItemReaderBuilder()
		.name(getNameFromPropertiesFile())
		// set other properties
		.build();
```

In this example, if `getNameFromPropertiesFile()` returns null or an empty string, the reader will be created in an inconsistent state. That's why it is better to fail fast at the builder level.